pandas/core/internals/managers.py

```diff
@@ -66,7 +68,618 @@
+        return self.reindex_indexer(
+            new_index, indexer, axis=axis, fill_value=fill_value, copy=copy
+        )
+
+
+class ArrayManager(DataManager):
+
+    __slots__ = [
+        "_axes",
+        "arrays",
+    ]
+
+    arrays: List[np.ndarray]
+    axes: Sequence[Index]
+
+    def __init__(
+        self,
+        arrays: List[np.ndarray],
+        axes: Sequence[Index],
+        do_integrity_check: bool = True,
+    ):
+        self._axes = axes
+        self.arrays = arrays
+
+        if do_integrity_check:
+            self._axes = [ensure_index(ax) for ax in axes]
+            self._verify_integrity()
+
+    @property
+    def items(self) -> Index:
+        return self._axes[1]
+
+    @property
+    def axes(self) -> Sequence[Index]:
+        return [self._axes[1], self._axes[0]]
+
+    @property
+    def shape(self) -> Tuple[int, ...]:
+        # this still gives the "old" transposed shape
+        return tuple(len(ax) for ax in self.axes)
+
+    @property
+    def shape_proper(self) -> Tuple[int, ...]:
+        # this returns (n_rows, n_columns)
+        return tuple(len(ax) for ax in self._axes)
+
+    @staticmethod
+    def _normalize_axis(axis):
+        # switch axis
+        axis = 1 if axis == 0 else 0
+        return axis
+
+    def make_empty(self: T, axes=None) -> T:
+        """ return an empty BlockManager with the items axis of len 0 """
+        if axes is None:
+            axes = [self.axes[1:], Index([])]
+
+        arrays = []
+        return type(self)(arrays, axes)
+
+    def consolidate(self) -> "ArrayManager":
+        return self
+
+    def is_consolidated(self) -> bool:
+        return True
+
+    def _consolidate_inplace(self) -> None:
+        pass
+
+    # TODO can be shared
+    def set_axis(self, axis: int, new_labels: Index) -> None:
+        # Caller is responsible for ensuring we have an Index object.
+        axis = self._normalize_axis(axis)
+        old_len = len(self._axes[axis])
+        new_len = len(new_labels)
+
+        if new_len != old_len:
+            raise ValueError(
+                f"Length mismatch: Expected axis has {old_len} elements, new "
+                f"values have {new_len} elements"
+            )
+
+        self._axes[axis] = new_labels
+
+    def get_dtypes(self):
+        return np.array([arr.dtype for arr in self.arrays], dtype="object")
+
+    # TODO setstate getstate
+
+    def __repr__(self) -> str:
+        output = type(self).__name__
+        output += f"\nIndex: {self._axes[0]}"
+        output += f"\nColumns: {self._axes[1]}"
+        output += f"\n{len(self.arrays)} arrays:"
+        for arr in self.arrays:
+            output += f"\n{arr.dtype}"
+        return output
+
+    def _verify_integrity(self) -> None:
+        pass
+        # TODO
+        # mgr_shape = self.shape
+        # tot_items = sum(len(x.mgr_locs) for x in self.blocks)
+        # for block in self.blocks:
+        #     if block._verify_integrity and block.shape[1:] != mgr_shape[1:]:
```

Comment:
i think this can be replaced with just checking that all of the array lengths match